sloglint.go

```diff
@@ -114,25 +114,34 @@ func flags(opts *Options) flag.FlagSet {
 	return *fset
 }
 
-var slogFuncs = map[string]int{ // funcName:argsPos
-	"log/slog.Log":                    3,
-	"log/slog.Debug":                  1,
-	"log/slog.Info":                   1,
-	"log/slog.Warn":                   1,
-	"log/slog.Error":                  1,
-	"log/slog.DebugContext":           2,
-	"log/slog.InfoContext":            2,
-	"log/slog.WarnContext":            2,
-	"log/slog.ErrorContext":           2,
-	"(*log/slog.Logger).Log":          3,
-	"(*log/slog.Logger).Debug":        1,
-	"(*log/slog.Logger).Info":         1,
-	"(*log/slog.Logger).Warn":         1,
-	"(*log/slog.Logger).Error":        1,
-	"(*log/slog.Logger).DebugContext": 2,
-	"(*log/slog.Logger).InfoContext":  2,
-	"(*log/slog.Logger).WarnContext":  2,
-	"(*log/slog.Logger).ErrorContext": 2,
+type slogFuncInfo struct {
+	argsPos          int
+	skipContextCheck bool
+}
+
+var slogFuncs = map[string]slogFuncInfo{ // funcName:
```

Comment:
Idk, probably not worth the expansion just for `With`. I'd replace `// NOTE: with functions are not checked for context.Context.` and `if !funcInfo.skipContextCheck` with a simple `if name != "With"` for now.

Everything else is LGTM.

Reply:
I would prefer to keep as is because `name` is `log/slog.With` or `(*log/slog.Logger).With` and not simply `With`.

I can check `!strings.HasSuffix(name, ".With")`  but it feels unnatural to me adding such check when we can just declare the function data with our intent - skip the context check. 

Let me know if you prefer the `HasSuffix` check instead of using the expansion

Reply:
Sure, it's just a minor detail, I'm fine with either.

Also, `slogFuncInfo` can be inlined as it is only used here.

Reply:
Done in 9bbec78355d6de1e88c769c131ca3415d02174c4